admin/query/src/main/java/org/codice/ditto/replication/admin/query/queries/GetQueries.java

```diff
@@ -0,0 +1,100 @@
+package org.codice.ditto.replication.admin.query.queries;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import java.util.List;
+import java.util.Set;
+import org.codice.ddf.admin.api.Field;
+import org.codice.ddf.admin.api.fields.FunctionField;
+import org.codice.ddf.admin.api.fields.ListField;
+import org.codice.ddf.admin.common.fields.base.BaseFunctionField;
+import org.codice.ditto.replication.api.QueryService;
+import org.codice.ditto.replication.api.data.Query;
+
+/**
+ * This is a graphql function field. A call to the graphql endpoint with the given field name will
+ * trigger the performFunction method and present the returned data in the response.
+ */
+public class GetQueries extends BaseFunctionField<ListField<QueryField>> {
+
+  private static final String FIELD_NAME = "queries";
+
+  private static final String DESCRIPTION =
+      "Retrieves all saved queries for the currently logged in user.";
+
+  private static final ListField<QueryField> RETURN_TYPE = new QueryField.ListImpl();
+
+  private final QueryService queryService;
+
+  /**
+   * Creates a new GetQueries function field.
+   *
+   * @param queryService A service which will fetch queries from the catalog framework for us
+   */
+  public GetQueries(QueryService queryService) {
+    super(FIELD_NAME, DESCRIPTION);
+    this.queryService = queryService;
+  }
+
+  /**
+   * Contains all the functionality of the "queries" graphql query. The response of the query will
+   * contain the information in the {@link QueryField.ListImpl}.
+   *
+   * @return the result of a query for saved user queries in the form of a {@link
+   *     QueryField.ListImpl}
+   */
+  @Override
+  public ListField<QueryField> performFunction() {
+    ListField<QueryField> queryFields = new QueryField.ListImpl();
+    queryService.queries().map(this::getFieldForQuery).forEach(queryFields::add);
```

Comment:
❓ Should we catch the ReplicationException?

Reply:
If not then you need to document it can be thrown out

Reply:
I think it would only make sense to catch it if we wanted to return an error message, and we only return an error message if there's something that can be done outside of the system to correct the issue. Since any issue that would cause the exception would have to be internal, I think we should just let the exception bubble out and have the graphql enpdoint return with the "internal server error" message.